### DISCUSSION TITLE
Where is the redirect rule path matching algorithm in the codebase?

### DISCUSSION THREAD

**Discussion by karpawich** [vercel/next.js#69421] [Help] [answered]
### Summary

Hello!!

I was looking for algorithm in the Next.js codebase that matches a given URL/path against the [redirect rules in the Next.js config](https://github.com/vercel/next.js/blob/d66ea6ad507678ade80f0aecdcbbc8ebc9b8e17e/examples/redirects/next.config.js), e.g.

```javascript
{
  source: "/old-blog/:slug",
  destination: "/news/:slug",
  permanent: false,
}
```

 but I was unsuccessful. Can anyone with more knowledge of the codebase help me? 🙏🏻

**Comment by Juneezee** [comment, marked as answer]
Hope it helps

https://github.com/vercel/next.js/blob/9720d0adc0b0687267e35804b6f6380fa351ab0c/packages/next/src/lib/load-custom-routes.ts#L577-L593

https://github.com/vercel/next.js/blob/9720d0adc0b0687267e35804b6f6380fa351ab0c/packages/next/src/lib/load-custom-routes.ts#L686-L693

https://github.com/vercel/next.js/blob/9720d0adc0b0687267e35804b6f6380fa351ab0c/packages/next/src/build/index.ts#L718-L728

**Comment by karpawich** [comment]
You are the best. Thank you so much @Juneezee 🙌🏻
